bundles/org.openhab.binding.clementineremote/proto/remotecontrolmessages.proto

```diff
@@ -0,0 +1,404 @@
+message RequestSetTrackPosition {
+    optional int32 position = 1;
+}
+
+message RequestInsertUrls {
+    // In which playlist should the urls be inserted?
+    optional int32 playlist_id = 1;
+    repeated string urls = 2;
+    optional int32 position = 3 [default=-1];
+    optional bool play_now = 4 [default=false];
+    optional bool enqueue = 5 [default=false];
+    repeated SongMetadata songs = 6;
+}
+
+// Client want to change track
+message RequestRemoveSongs {
+    // In which playlist is the songs?
+    optional int32 playlist_id = 1;
+    // And on which position?
+    repeated int32 songs = 2;
+}
+
+// Messages for opening / closing playlists
+message RequestOpenPlaylist {
+    optional int32 playlist_id = 1;
+}
+message RequestClosePlaylist {
+    optional int32 playlist_id = 1;
+}
+
+// Message containing lyrics
+message ResponseLyrics {
+    repeated Lyric lyrics = 1;
+}
+message Lyric {
+    optional string id = 1;
+    optional string title = 2;
+    optional string content = 3;
+}
+
+// Message request for downloading songs
+enum DownloadItem {
+    CurrentItem = 1;
+    ItemAlbum = 2;
+    APlaylist = 3;
+    Urls = 4;
+}
+message RequestDownloadSongs {
+    optional DownloadItem download_item = 1;
+    optional int32 playlist_id = 2;
+    repeated string urls = 3;
+}
+
+message ResponseSongFileChunk {
+    optional int32 chunk_number = 1;
+    optional int32 chunk_count = 2;
+    optional int32 file_number = 3;
+    optional int32 file_count = 4;
+    optional SongMetadata song_metadata = 6; // only sent with first chunk!
+    optional bytes data = 7;
+    optional int32 size = 8;
+    optional bytes file_hash = 9;
+}
+
+message ResponseLibraryChunk {
+    optional int32 chunk_number = 1;
+    optional int32 chunk_count = 2;
+    optional bytes data = 3;
+    optional int32 size = 4;
+    optional bytes file_hash = 5;
+}
+
+message ResponseSongOffer {
+    optional bool accepted = 1; // true = client wants to download item
+}
+
+message RequestRateSong {
+    optional float rating = 1; // 0 to 1
+}
+
+message ResponseDownloadTotalSize {
+    optional int32 total_size = 1;
+    optional int32 file_count = 2;
+}
+
+message RequestGlobalSearch {
+    optional string query = 1;
+}
+
+message ResponseGlobalSearch {
+    optional int32 id = 1;
+    optional string query = 2;
+    optional string search_provider = 3;
+    repeated SongMetadata song_metadata = 4;
+    optional bytes search_provider_icon = 5;
+}
+
+message ResponseTranscoderStatus {
+    optional int32 processed = 1;
+    optional int32 total = 2;
+}
+
+enum GlobalSearchStatus {
+    GlobalSearchStarted = 1;
+    GlobalSearchFinished = 2;
+}
+
+message ResponseGlobalSearchStatus {
+    optional int32 id = 1;
+    optional string query = 2;
+    optional GlobalSearchStatus status = 3;
+}
+
+// The message itself
+message Message {
+    optional int32 version = 1 [default=21];
+    optional MsgType type = 2 [default=UNKNOWN]; // What data is in the message?
```

Comment:
Is this meant as some todo? Isf so, please comment as TODO:

Reply:
I think this is a developer hint. The file originates from the Clementine Android-Remote project. As such, I don't want to alter it.